foreign_cc/private/runnable_binary_wrapper.sh

```diff
@@ -4,9 +4,20 @@
 # Otherwise, first cd to the runfiles dir for the wrapped executable before searching for shared libraries for the wrapped executable
 if [[ -z $RUN_UNDER_RUNFILES ]]; then
     SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
-    RUNFILES_DIR=${SCRIPT_DIR}/SH_BINARY_FILENAME.runfiles
+    RUNFILES_DIR_TMP=${SCRIPT_DIR}/SH_BINARY_FILENAME.runfiles
```

Comment:
Under what conditions do the runfiles not exist? I have to admit I'm not familiar enough with the details of runfile layout to know if this is correct or not.

@fmeum is this logic correct or are we missing something else runfiles related here?

Reply:
I think that this logic isn't sound, but it also wasn't sound before the PR (not entirely sure though, this is genuinely tricky). `RUNFILES_DIR` must always take precedence if already set, but this logic appears to overwrite based in whether the current script has a runfiles directory, which can result in non-hermetic behavior.

Since the script uses the runfiles library below, could that setup code be moved up? That would provide the correct variables and this script wouldn't have to worry about the details. I also don't see a need to have a special case for `--run_under`.

Reply:
Thank you for taking a look, I've just removed the special case related to `--run_under`. Since we don't use `--run_under`, I'm not sure if that preserves other users workflows (but it looks like they should continue working)

> Under what conditions do the runfiles not exist?

I've found that using the previous logic would fail if the runnable binary was a data input to another binary. The rules_foreign_cc binary would be present in the runfiles directory, and wouldn't have a runfiles directory itself (since all the runfiles were aggregated in a top level directory)

Reply:
That makes sense. If a binary is a data dependency of some other binary, the outer binary has to use the runfiles libraries to set `RUNFILES_DIR`, which the inner binary can then read and use.